dotnet/src/webdriver/Proxy.cs

```diff
@@ -93,93 +95,90 @@ public Proxy()
             {
                 throw new ArgumentNullException(nameof(settings), "settings dictionary cannot be null");
             }
 
-            if (settings.ContainsKey("proxyType") && settings["proxyType"] != null)
+            if (settings.TryGetValue("proxyType", out object? proxyTypeObj) && proxyTypeObj?.ToString() is string proxyType)
             {
                 // Special-case "PAC" since that is the correct serialization.
-                if (settings["proxyType"].ToString().ToLowerInvariant() == "pac")
+                if (proxyType.Equals("pac", StringComparison.InvariantCultureIgnoreCase))
                 {
                     this.Kind = ProxyKind.ProxyAutoConfigure;
                 }
                 else
                 {
-                    ProxyKind rawType = (ProxyKind)Enum.Parse(typeof(ProxyKind), settings["proxyType"].ToString(), true);
+                    ProxyKind rawType = Enum.Parse<ProxyKind>(proxyType, ignoreCase: true);
                     this.Kind = rawType;
                 }
             }
 
-            if (settings.ContainsKey("ftpProxy") && settings["ftpProxy"] != null)
+            if (settings.TryGetValue("ftpProxy", out object? ftpProxyObj) && ftpProxyObj?.ToString() is string ftpProxy)
             {
-                this.FtpProxy = settings["ftpProxy"].ToString();
+                this.FtpProxy = ftpProxy;
             }
 
-            if (settings.ContainsKey("httpProxy") && settings["httpProxy"] != null)
+            if (settings.TryGetValue("httpProxy", out object? httpProxyObj) && httpProxyObj?.ToString() is string httpProxy)
             {
-                this.HttpProxy = settings["httpProxy"].ToString();
+                this.HttpProxy = httpProxy;
             }
 
-            if (settings.ContainsKey("noProxy") && settings["noProxy"] != null)
+            if (settings.TryGetValue("noProxy", out object? noProxy) && noProxy != null)
             {
                 List<string> bypassAddresses = new List<string>();
-                string addressesAsString = settings["noProxy"] as string;
-                if (addressesAsString != null)
+                if (noProxy is string addressesAsString)
                 {
                     bypassAddresses.AddRange(addressesAsString.Split(';'));
                 }
-                else
+                else if (noProxy is object?[]  addressesAsArray)
                 {
-                    object[] addressesAsArray = settings["noProxy"] as object[];
-                    if (addressesAsArray != null)
+                    foreach (object? address in addressesAsArray)
                     {
-                        foreach (object address in addressesAsArray)
-                        {
-                            bypassAddresses.Add(address.ToString());
-                        }
+                        bypassAddresses.Add(address?.ToString() ?? throw new ArgumentException("noProxy list contained null element", nameof(settings)));
```

Comment:
This message in exception is unclear for end-user. Is ther nay way to improve it?

Reply:
In order to reach this exception, the user would have to do something like this:

```csharp
var proxyValues = new Dictionary<string, object?>()
{
    ["noProxy"] = new string?[] { null }
};

var proxy = new Proxy(proxyValues);
```

Which makes the `noProxy` name is user-facing. What about a message like `"Proxy bypass address list \"noProxy\" contained a null element."`?

Reply:
Minor, but it is better, considering your sample.

Reply:
Fixed